test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1347,10 +1336,19 @@ private synchronized void startAndPublishNodesAndClients(List<NodeAndClient> nod
                 // special case for 1 node master - we can't update the min master nodes before we add more nodes.
                 updateMinMasterNodes(currentMasters + newMasters);
             }
-            for (NodeAndClient nodeAndClient : nodeAndClients) {
-                nodeAndClient.startNode();
-                publishNode(nodeAndClient);
+            List<Future<?>> futures = nodeAndClients.stream().map(node -> executor.submit(node::startNode)).collect(Collectors.toList());
+            try {
+                for (Future<?> future : futures) {
+                    future.get();
+                }
+            } catch (InterruptedException e) {
+                Thread.interrupted();
+                return;
```

Comment:
I wonder about the cases where this can happen as the `close` method on `InternalTestCluster` that calls `shutdown` on the executor is synchronized (same as this method). Have you observed this exception being thrown? If we don't expect this to occur under normal operations, I would prefer not to swallow the exception here.